core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,227 @@
+			}
+
+			DestTokenAddress, err := solana.PublicKeyFromBase58(string(tokenAmount.DestTokenAddress))
+			if err != nil {
+				return nil, fmt.Errorf("invalid destTokenAddress address: %w", err)
+			}
+
+			tokenAmounts = append(tokenAmounts, ccip_router.Any2SolanaTokenTransfer{
+				SourcePoolAddress: tokenAmount.SourcePoolAddress,
+				DestTokenAddress:  DestTokenAddress,
+				ExtraData:         tokenAmount.ExtraData,
+				Amount:            bigIntToBytes32(tokenAmount.Amount),
+				DestGasAmount:     bytesToUint32(tokenAmount.DestExecData),
+			})
+		}
+
+		var extraArgs ccip_router.SolanaExtraArgs
+		decoder := agbinary.NewBorshDecoder(msg.ExtraArgs)
+		err := extraArgs.UnmarshalWithDecoder(decoder)
+		if err != nil {
+			return nil, fmt.Errorf("invalid extra arguments: %w", err)
+		}
+
+		receiver, err := solana.PublicKeyFromBase58(string(msg.Receiver))
+		if err != nil {
+			return nil, fmt.Errorf("invalid receiver address: %s, %w", string(msg.Receiver), err)
+		}
+
+		message = ccip_router.Any2SolanaRampMessage{
+			Header: ccip_router.RampMessageHeader{
+				MessageId:           msg.Header.MessageID,
+				SourceChainSelector: uint64(msg.Header.SourceChainSelector),
+				DestChainSelector:   uint64(msg.Header.DestChainSelector),
+				SequenceNumber:      uint64(msg.Header.SequenceNumber),
+				Nonce:               msg.Header.Nonce,
+			},
+			Sender:       msg.Sender,
+			Data:         msg.Data,
+			Receiver:     receiver,
+			TokenAmounts: tokenAmounts,
+			ExtraArgs:    extraArgs,
+		}
+	}
+
+	var offchainTokenData [][]byte
+	if len(chainReport.OffchainTokenData) > 0 {
+		offchainTokenData = chainReport.OffchainTokenData[0]
+	}
+
+	solanaProofs := make([][32]byte, 0, len(chainReport.Proofs))
+	for _, proof := range chainReport.Proofs {
+		solanaProofs = append(solanaProofs, proof)
+	}
+
+	solanaReport := ccip_router.ExecutionReportSingleChain{
+		SourceChainSelector: uint64(chainReport.SourceChainSelector),
+		Message:             message,
+		OffchainTokenData:   offchainTokenData,
+		Proofs:              solanaProofs,
+		TokenIndexes:        calculateTokenIndexes(message),
+	}
+
+	var buf bytes.Buffer
+	encoder := agbinary.NewBorshEncoder(&buf)
+	err := solanaReport.MarshalWithEncoder(encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (e *ExecutePluginCodecV1) Decode(ctx context.Context, encodedReport []byte) (cciptypes.ExecutePluginReport, error) {
+	decoder := agbinary.NewBorshDecoder(encodedReport)
+	executeReport := ccip_router.ExecutionReportSingleChain{}
+	err := executeReport.UnmarshalWithDecoder(decoder)
+	if err != nil {
+		return cciptypes.ExecutePluginReport{}, fmt.Errorf("unpack encoded report: %w", err)
+	}
+
+	tokenAmounts := make([]cciptypes.RampTokenAmount, 0, len(executeReport.Message.TokenAmounts))
+	for _, tokenAmount := range executeReport.Message.TokenAmounts {
+		destData := make([]byte, 4)
+		binary.BigEndian.PutUint32(destData, tokenAmount.DestGasAmount)
```

Comment:
Again, I'm a little lost on what this encoding should be like.
Need someone from CCIP to explain to us.